processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -685,6 +692,15 @@ public int size()
     return numEntries.get();
   }
 
+  public int estimatedOccupation()
+  {
+    int occupation = maxLengthForAggregators * getFacts().size();
```

Comment:
I agree on all of the your comments. It's just for estimation which can be relative (if it's half of real occupation, we just can set it as a half) and advisory at best rather than exact. 

- for "maxLengthForAggregators", I'm just preying it's similar to on heap aggregator (Or should we add a method to AggregatorFactory?)
- for "dimension estimation", would it be enough to add Ints.BYTES for occupation of index?
- for "underestimate" part. Yes, I've thought of that and resulted rather small value of "defaultMaxOccupationInMemory", that you've commented on next of this
- for "IncrementalIndex implementation" part, I don't think it's critically different between current implementations on memory usage and also we are only using on-heap one for realtime indexing. But ideally it's the best way things to be done, I think.